core/src/main/java/org/apache/calcite/rel/rules/ProjectToWindowRule.java

```diff
@@ -246,8 +251,17 @@ protected RelNode makeRel(RelOptCluster cluster, RelTraitSet traitSet,
               RelBuilder relBuilder, RelNode input, RexProgram program) {
             Preconditions.checkArgument(program.getCondition() == null,
                 "WindowedAggregateRel cannot accept a condition");
-            return LogicalWindow.create(cluster, traitSet, relBuilder, input,
-                program);
+            List<RelCollation> collations
+                = traitSet.getTraits(RelCollationTraitDef.INSTANCE);
+            RelNode window = LogicalWindow.create(cluster, relBuilder, input, program);
+            if (collations == null || collations.isEmpty() || collations.size() != 1) {
+              return window;
+            }
+            RelCollation collation = collations.get(0);
+            if (Objects.equals(collation, RelCollations.EMPTY)) {
+              return window;
+            }
+            return LogicalSort.create(window, collation, null, null);
```

Comment:
What is the point of manually creating `LogicalSort`?

Reply:
If LogicalProject generates collations, then its parent rel could probably asks for a Window rel with collation. In this case a explicit sort is needed to make the window match the order.